pallets/file-system/src/utils.rs

```diff
@@ -1196,49 +1174,62 @@ where
         // Return the threshold needed for the BSP to be able to volunteer for the storage request.
         T::HashToThresholdType::convert(volunteering_hash)
     }
-}
 
-impl<T: crate::Config> shp_traits::SubscribeProvidersInterface for Pallet<T> {
-    type ProviderId = ProviderIdFor<T>;
+    pub fn compute_threshold_to_succeed(
+        bsp_id: &ProviderIdFor<T>,
+        requested_at: BlockNumberFor<T>,
+    ) -> Result<(T::ThresholdType, T::ThresholdType), DispatchError> {
+        let maximum_threshold: u32 = MaximumThreshold::<T>::get().into();
```

Comment:
This essentially makes the `MaximumThreshold` type useless 😂

Reply:
Thanks for point this out along with the other related comments regarding the use of `u32`.